src/config.cc

```diff
@@ -93,6 +93,7 @@ void NetworkConfig::updateFromPropertyTree(const boost::property_tree::ptree& pt
   CopyFromConfig(ipdiscovery_host, "ipdiscovery_host", boost::log::trivial::trace, pt);
   CopyFromConfig(ipdiscovery_port, "ipdiscovery_port", boost::log::trivial::trace, pt);
   CopyFromConfig(ipdiscovery_wait_seconds, "ipdiscovery_wait_seconds", boost::log::trivial::trace, pt);
+  CopyFromConfig(ipuptane_port, "network.ipuptane_port", boost::log::trivial::trace, pt);
```

Comment:
Same idea as here: https://github.com/advancedtelematic/aktualizr/pull/632#discussion_r174523296

By returning the port in the discovery packet, we would avoid another possibly misconfigured config item.

Reply:
Sorry, didn't notice that. Yes, totally makes sense, I'll add it to this PR.

Reply:
And since the communication is now bidirectional, the primary's tcp port could be in the discovery request?

Reply:
@lbonn I've also found a fatal flow in the discovery: it doesn't actually save discovered IP addresses =)